wdc-calculator-core/wdc-calculator-core-action/src/main/java/hu/icellmobilsoft/wdc/calculator/core/action/WorkdayCacheInitializer.java

```diff
@@ -53,15 +69,63 @@
     public void init(@Observes @Initialized(ApplicationScoped.class) Object init) {
         this.initWorkdayCache();
         try {
-            workdayDataReader.readAll();
-        } catch (BusinessException e) {
+            initDirWatch();
+        } catch (IOException e) {
             throw new RuntimeException(e.getMessage(), e);
         }
+
+        initScheduler();
     }
 
     private void initWorkdayCache() {
-        workdayCache.initYear(Year.now().minusYears(1));
-        workdayCache.initYear(Year.now());
-        workdayCache.initYear(Year.now().plusYears(1));
+        synchronized (workdayCache.getLockObject()) {
+            workdayCache.clearCache();
+
+            workdayCache.initYear(Year.now().minusYears(1));
+            workdayCache.initYear(Year.now());
+            workdayCache.initYear(Year.now().plusYears(1));
+
+            try {
+                workdayDataReader.readAll();
+            } catch (BusinessException e) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
+        }
+    }
+
+    private void initScheduler() {
+        Timer timer = new Timer();
+        TimerTask task = new TimerTask() {
+            @Override
+            public void run() {
+                checkChanges();
+            }
+        };
+        timer.scheduleAtFixedRate(task, 0, 3600000);
+    }
+
+    public void checkChanges() {
```

Comment:
It's nice to have, as some uses might want to update more frequently.
But i forgot javadoc for it, will fix that in a moment.

Reply:
Now only used by initScheduler() method within class, so private modifier more reasonable. It can be changed to public if necessary in the future (if it will be necessary at all).